packages/extension-ui/src/Popup/Accounts/ChangePassword.tsx

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019-2023 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useCallback, useContext, useState } from 'react';
+import { useParams } from 'react-router';
+import styled from 'styled-components';
+
+import { getUserInputs } from '@polkadot/extension-ui/components/PasswordField/getFeedback';
+import Message from '@polkadot/extension-ui/components/PasswordField/Message';
+import useAccountName from '@polkadot/extension-ui/hooks/useAccountName';
+
+import {
+  ActionContext,
+  Address,
+  Button,
+  ButtonArea,
+  InputWithLabel,
+  ScrollWrapper,
+  ValidatedInput,
+  VerticalSpace
+} from '../../components';
+import useToast from '../../hooks/useToast';
+import useTranslation from '../../hooks/useTranslation';
+import { changePassword } from '../../messaging';
+import { Header, Password } from '../../partials';
+import { Result } from '../../util/validators';
+
+
+const EditPassword = () => {
+  const { t } = useTranslation();
+  const { show } = useToast();
+
+  const { address } = useParams<{ address: string }>();
+
+  const onAction = useContext(ActionContext);
+  const goBack = () => onAction('..');
+
+  const accountName = useAccountName(address);
+
+  const [providedPass, setProvidedPass] = useState('');
+  const [isProvidedPassWrong, setIsProvidedPassWrong] = useState(false);
+  const [nextPass, setNextPass] = useState<string | null>(null);
+
+  const saveChanges = async (): Promise<void> => {
+    if (nextPass === null) {
+      return;
+    }
+
+    try {
+      await changePassword(address, providedPass, nextPass);
+      goBack();
+      show(t<string>('Account password changed successfully'), 'success');
+    } catch (error) {
+      show(t<string>('Unable to decode using the supplied passphrase'), 'critical');
```

Comment:
Maybe more user-friendly:
```suggestion
      show(t<string>('The current password is invalid'), 'critical');
```
or sth like that?